### PR TITLE
Add the flag is_ethflow_order

### PR DESCRIPTION
**Pull request by josojo**
Adds a new flag `is_ethflow_order`. This flag is required, as we need to do some special operations for ethflow orders, e.g. add a special pre_interaction to wrap eth.

Please review together with the follow-up: [PR](https://github.com/cowprotocol/services/pull/581)

Note: There is an alternative way to implement it:
- storing the ethflow_address in the settlement_encoder and then identifying ethflow orders as orders that use this ethflow_address as owner. However, I found this more cumbersome for 1 main reason: This would mean we need to propagate the eth_flow address to all places where we need to check whether an order is an ethflow order. Just alone for the settlement encoder, this requires storing the ethflow_address in the SettlementEncoder + Settlement structs. Since they are currently created by the solvers, this would affect all solver implementation, unless it's nicely abstracted away by a new abstraction layer.

### Test Plan
unit test

### CODE REVIEW

File: crates/database/src/orders.rs [hunk 1]
```diff
@@ -174,8 +174,8 @@ pub async fn read_order(
     id: &OrderUid,
 ) -> Result<Option<Order>, sqlx::Error> {
     const QUERY: &str = r#"
-SELECT * FROM ORDERS
-WHERE uid = $1
+        SELECT * FROM ORDERS
+        WHERE uid = $1
```
Comment (vkgnosis) [inline crates/database/src/orders.rs:178]: I had this indented to the left on purpose so larger sql statements become more readable. But doesn't really matter.

**Review by vkgnosis — APPROVED** [review]

**Review by nlordell — APPROVED** [review]
LGTM.

**Comment by nlordell** [comment]
> this requires storing the ethflow_address in the SettlementEncoder + Settlement structs. Since they are currently created by the solvers, this would affect all solver implementation, unless it's nicely abstracted away by a new abstraction layer.

Looking at this with the follow up PR - why not add the "ethflow order identification" to the `solver::liquidity::order_converter::OrderConverter` and have it add an `is_ethflow_order` flag to the `LimitOrder` type instead of extending the order data?

Then, the `SettlementEncoder` implementation can add the ethflow contract call as a pre-interaction? I think this may be less intrusive than making the orderbook + auction `is_ethflow_order` aware, and just have the code-smell in the driver to deal with until we support user-specified "pre-interactions".

**Comment by josojo** [comment]
@nlordell, sorry, I should have linked this PR as well: https://github.com/cowprotocol/services/pull/588

The autopilot also needs custom logic for cutting orders for ethflow orders. Generally, I like your proposal to limit the smell only to the driver, but at least the smell also needs to get into the autopilot. 
I have not yet fully thought through the orderbook API interactions. Maybe the smell even needs to be in the orderbook API: Maybe we wanna do a better gas estimation than on regular EIP1271 orders with less buffers.

Given these circumstances, should I still implement your change?

**Comment by nlordell** [comment]
> Given these circumstances, should I still implement your change?

No, I think its fine then.

**Review by nlordell — APPROVED** [review]

**Comment by MartinquaXD** [comment]
> Just alone for the settlement encoder, this requires storing the ethflow_address in the SettlementEncoder + Settlement structs.

Why do we need to propagate those addresses through the program? Adresses of contracts with specialized handling (like ETH flow) are constant so we could simply have a hard coded list in the binary. This list would have a static lifetime and is therefore trivial to access throughout the program.

**Pull request merged**